### ISSUE TITLE
Issues with the predictions

### ISSUE THREAD

**Issue by rubenaros** [Irene323/GFTE#16]
Hi,  i implemented a versión of prediction, but the results are bads. Can you help me? 

def predict(model,structs, chunks, imgfn):
    cl = cal_chk_limits(chunks)
    x,pos,tbpos,xtext,imgpos=[],[],[],[],[]
    plaintext = []
    structs = remove_empty_cell(structs)
    for st in structs:
        id = st["id"] 
        chk = chunks[id]
        xt = pos_feature(chk,cl)
        x.append(xt)
        pos.append(xt[4:6])
        xtext.append(encode_text(chk["text"],vob))
        plaintext.append(chk["text"].encode('utf-8'))
        imgpos.append([(1.0-xt[5])*2-1.0, xt[4]*2-1.0]) # 图像中的y是倒过来的。这是归一化[-1,1]之间。图像的y在前，和H对应。          

    image = cv2.imread(imgfn)
    img_size = 256
    kernel = np.ones((3,3),np.uint8)  # 把图像的线变粗一点

    imagetmp = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
    plt.imshow(imagetmp)
    plt.show()

    if not image is None:
       image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
       image = cv2.dilate(image,kernel,iterations = 1)
       image = cv2.resize(image, (img_size,img_size), interpolation = cv2.INTER_AREA)          
            
    x = torch.FloatTensor(x)  
    pos = torch.FloatTensor(pos)  
    data = Data(x=x,pos=pos)
    d = GT.KNNGraph(k=6)
    data = d(data)
    image = torch.FloatTensor(image/255.0).unsqueeze(0).unsqueeze(0)
    #print(img.size(), img.dtype)
    data.img = image
    data.imgpos = torch.FloatTensor(imgpos)
    data.nodenum = torch.LongTensor([len(structs)])
    # print(type(xtext)) #<class 'list'>
    data.xtext = torch.LongTensor(xtext)    
    print(data.nodenum)
    print(data.x)

    model.eval()
    with torch.no_grad():
         #data = data.cuda()
         preds = model(data)
         _, preds1 = preds.max(1)
    print(preds)
    print(preds1)
    points = [ (x,y) for x,y in zip(data.edge_index[0], data.edge_index[1])]
    l = len(preds1)
    index = [x for x in range(l) if preds1[x]==1]
    points = [(data.edge_index[0][i].detach().cpu().item(),data.edge_index[1][i].detach().cpu().item()) for i in index]
    points = sort_tuple(points)
    print_connected_points(points,chunks)
    return

#######################################################################################################

nclass = 2
input_num = 8
vocab_size = 39
num_text_features = 64
test_path = "D:/GitHub/SciTSR/scitsr/data/SciTSR/test"

print('num of classes:',nclass)
